src/annotations/components/EditAnnotationForm.tsx

```diff
@@ -46,32 +47,45 @@ export const EditAnnotationForm: FC<Props> = (props: Props) => {
     id: props.annotation.id,
     message: props.annotation.message ?? '',
     startTime: new Date(props.annotation.startTime).toISOString(),
+    endTime: new Date(props.annotation.endTime).toISOString(),
     stream: props.annotation.stream,
     summary: props.annotation.summary,
+    type:
+      props.annotation.startTime === props.annotation.endTime
+        ? 'point'
+        : 'range',
   })
 
   const dispatch = useDispatch()
 
-  const isValidAnnotationForm = ({summary, startTime}): boolean => {
-    return summary.length && startTime
+  const isValidAnnotationForm = (): boolean => {
+    return checkAnnotationFormValidity(
+      editedAnnotation.type,
+      editedAnnotation.summary,
+      editedAnnotation.startTime,
+      editedAnnotation.endTime
+    )
   }
 
-  const updateStartTime = (newStartTime: string) => {
+  const doUpdate = (obj: any) => {
```

Comment:
something about this being called `doUpdate` feels weird. Usually having to name a method `doSomething` because `something`  is already used is a smell - maybe something is too abstract because we already have `updateAnnotation`? Or maybe it means this should be called `updateAnnotation` and the other one should be renamed to `updateAnnotationState`?

And calling an argument `obj` actually mildly offends me when a senior engineer does it. Come on, it's pretty obviously the annotation we're editing - so can you name it something like that please?